server/src/http/jwt/middleware.rs

```diff
@@ -0,0 +1,67 @@
+use crate::http::jwt::json_web_token::Identity;
+use crate::http::state::AppState;
+use axum::{
+    extract::State,
+    http::{Request, StatusCode},
+    middleware::Next,
+    response::Response,
+};
+use std::sync::Arc;
+
+const AUTHORIZATION: &str = "authorization";
+const BEARER: &str = "Bearer ";
+const UNAUTHORIZED: StatusCode = StatusCode::UNAUTHORIZED;
+
+pub async fn jwt_auth<T>(
```

Comment:
why not
```rust

pub async fn jwt_auth(
    State(state): State<Arc<AppState>>,
    mut request: Request<_>,
    next: Next<_>,
)
```
isn't type inferred from Request<T>?

Reply:
It won't work/